src/ascent/runtimes/expressions/ascent_blueprint_architect.hpp

```diff
@@ -111,18 +111,32 @@ conduit::Node field_entropy(const conduit::Node &hist);
 conduit::Node field_pdf(const conduit::Node &hist);
 conduit::Node field_cdf(const conduit::Node &hist);
 
+conduit::Node global_bounds(const conduit::Node &dataset,
+                            const conduit::Node &field_names);
+
+conduit::Node binning(const conduit::Node &dataset,
+                  conduit::Node &bin_axes,
+                  const std::string &reduction_var,
+                  const std::string &reduction_op,
+                  const double empty_bin_val);
+
+void ASCENT_API paint_binning(const conduit::Node &binning, conduit::Node &dataset);
+
+conduit::Node ASCENT_API binning_mesh(const conduit::Node &binning);
```

Comment:
should the output be passed in as a ref? (If not the entire result will be copied)

Reply:
Yes I will update that.